common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -13,44 +13,49 @@
 import java.util.regex.Pattern;
 
 import static com.gradle.Utils.appendIfMissing;
-import static com.gradle.Utils.envVariable;
 import static com.gradle.Utils.execAndCheckSuccess;
 import static com.gradle.Utils.execAndGetStdOut;
-import static com.gradle.Utils.firstSysPropertyKeyStartingWith;
 import static com.gradle.Utils.isNotEmpty;
-import static com.gradle.Utils.projectProperty;
-import static com.gradle.Utils.readPropertiesFile;
 import static com.gradle.Utils.stripPrefix;
-import static com.gradle.Utils.sysProperty;
 import static com.gradle.Utils.urlEncode;
 
 /**
  * Adds a standard set of useful tags, links and custom values to all build scans published.
  */
 final class CustomBuildScanEnhancements {
 
-    static void apply(BuildScanExtension buildScan, ProviderFactory providers, Gradle gradle) {
-        captureOs(buildScan, providers);
-        captureIde(buildScan, providers, gradle);
-        captureCiOrLocal(buildScan, providers);
-        captureCiMetadata(buildScan, providers, gradle);
-        captureGitMetadata(buildScan);
-        captureTestParallelization(buildScan, gradle);
+    private final BuildScanExtension buildScan;
+    private final ProviderFactory providers;
+    private final Gradle gradle;
+
+    CustomBuildScanEnhancements(BuildScanExtension buildScan, ProviderFactory providers, Gradle gradle) {
+        this.buildScan = buildScan;
+        this.providers = providers;
+        this.gradle = gradle;
+    }
+
+    void apply() {
+        captureOs();
+        captureIde();
+        captureCiOrLocal();
+        captureCiMetadata();
+        captureGitMetadata();
+        captureTestParallelization();
     }
 
-    private static void captureOs(BuildScanExtension buildScan, ProviderFactory providers) {
```

Comment:
these were previously all static methods because of an issue being compatible with the configuration cache. I think this should just work now that they are members on the class but we should check before merging.

Reply:
Good suggestion. It looks like these changes do cause a problem with the configuration cache. I'm exploring options.

Reply:
Configuration cache fix has been added to the PR.